### PR TITLE
proof_level=check

### PR DESCRIPTION
**Pull request by cmr**
This implements a new mode, somewhere between `with_snarks=false` and `with_snarks=true`. We mostly turn `with_snarks` off for tighter debug cycles, but checking that the constraints hold shouldn't be that much slower than doing nothing (a later profile should test this), and will help us avoid long debug cycles with full proof computation.

### CODE REVIEW

**Comment by cmr** [comment]
On my machine, just doing `check` takes around 4-5s. generating a proof takes 9-12s.

**Comment by cmr** [comment]
This PR currently switches everything except `fake_hash` to the `check` level, which is slower. Given that it only cuts the proof time by a factor of two, this seems like it might be overly optimistic. Thoughts? Concretely, I turned the testnet_postake_snarkless block duration up to 6s.

File: src/app/cli/src/coda.ml [hunk 1]
```diff
@@ -405,13 +405,15 @@ let ensure_testnet_id_still_good _ = Deferred.unit
...
 [%%if
-with_snark]
+proof_level = "full"]
 
 let internal_commands =
   [(Snark_worker_lib.Intf.command_name, Snark_worker_lib.Prod.Worker.command)]
 
 [%%else]
 
+(* TODO: proof_level=check *)
```
Comment (bkase) [inline src/app/cli/src/coda.ml]: are these todos unnecessary now?

File: src/app/cli/src/coda_main.ml [hunk 2]
```diff
@@ -1008,6 +1010,7 @@ end
 
 [%%else]
 
+(* TODO: proof_level=check ledger proofs *)
```
Comment (bkase) [inline src/app/cli/src/coda_main.ml]: and this

File: src/app/cli/src/coda.ml (hunk 1 above)
Comment (cmr) [reply src/app/cli/src/coda.ml]: no, i added issue numbers

**Pull request merged**